tools/wptserve/wptserve/pipes.py

```diff
@@ -490,7 +491,11 @@ def config_replacement(match):
 
         #Should possibly support escaping for other contexts e.g. script
         #TODO: read the encoding of the response
-        return escape_func(text_type(value)).encode("utf-8")
+        if isinstance(value, binary_type):
+            value = value.decode("utf-8")
```

Comment:
~~Can we just avoid the roundtrip here?~~ No, we can't, can you add a comment saying *why* we can't? (i.e., the type of escape_func)